raiden/network/blockchain_service.py

```diff
@@ -212,6 +212,6 @@ def payment_channel(
         return self.identifier_to_payment_channel[dict_key]
 
     @property
-    @ttl_cache(ttl=30)
+    @ttl_cache(ttl=300)  # Check if the underlying eth network changed every 5 mins
     def network_id(self) -> int:
         return int(self.client.web3.version.network)
```

Comment:
My question here is that, and according to the linked conversation:
Would  the network/chain_id change while Raiden is running? if the assumption is that the user restarts geth/parity and switches chains then Raiden would have crashed and would need to be restarted. Unless i am missing something here.

Reply:
This is already done [here](https://github.com/raiden-network/raiden/blob/b538252715efac0dd3afae67923cd79542fe8a2f/raiden/tasks.py#L159-L163)

Reply:
Okay, so why don't we take the version once and just keep it around rather than getting the ID every 5 minutes?

Reply:
Because we check every 5 minutes if it has changed and then crash as per the linked code. Otherwise the value is cached. The effect is the same.

Reply:
Errr, why? I was pointing it out as a problem, we have had difficult to find bugs in the past because of unnecessary caches before, can we at very least measure to see why we are doing any of this?

Reply:
> Errr, why?

RPC calls are expensive when done frequently.

> Can we at very least measure to see why we are doing any of this?

Sure. The `network_id` property is called very often. Multple times during the handling of each block state change and during the alarm task. Also inside the `loop_until_stop`

To give you a rough idea of how often, this is how it looks like when I remove the cache completely and make it print before doing the RPC call.

![2019-02-06-153322_893x989_scrot](https://user-images.githubusercontent.com/1658405/52348580-f50d7580-2a24-11e9-91dd-3832cbfc6624.png)


----

If you don't like the 5 mins caching then I would go with what @rakanalh suggested. Remove the caching completely and instead save network_id as a property. Then have some kind of periodic task (perhaps along with the gas reserve checks) that performs the RPC call and checks if anything changed and if it did just crash with the proper error.

Now that I wrote that down .... this sounds even cleaner ...

Reply:
alternate suggestion PR coming soon